src/providers/wfs/qgswfsprovider.cpp

```diff
@@ -852,6 +852,16 @@ bool QgsWFSProvider::addFeatures( QgsFeatureList &flist )
   {
     //transaction successful. Add the features to the cache
     QStringList idList = insertedFeatureIds( serverResponse );
+    /* Fix issue with GeoServer and shapefile feature stores when no real
+       feature id are returned but new0 returned instead of the featureId*/
+  for ( const QString &v : idList )
```

Comment:
the `const QString & v` doesn't help - you could make idList const though:

    const QStringList idList = ...

See https://www.kdab.com/goodbye-q_foreach/